src/legacy/router.rs

```diff
@@ -161,15 +161,19 @@ export type Procedures = {{
 }};"#
         )?;
 
-        for export in self.type_map.iter().map(|(_, ty)| {
-            datatype(
-                &config,
-                &FunctionResultVariant::Value(ty.inner.clone()),
-                &self.type_map,
+        // Generate type exports (non-Procedures)
+        for (name, export) in self.type_map.iter().map(|(_, ty)| {
+            (
+                ty.name(),
+                datatype(
+                    &config,
+                    &FunctionResultVariant::Value(ty.inner.clone()),
+                    &self.type_map,
+                )
+                .unwrap(),
             )
-            .unwrap()
         }) {
-            writeln!(file, "\n{}", export)?;
+            writeln!(file, "\nexport type {} = {};", name, export)?;
```

Comment:
After looking into the [export implementation](https://github.com/specta-rs/specta/blob/13c74804f510d4002c58750717e6c532effdfa87/specta-typescript/src/typescript.rs#L148) of the `special-typescript` creation, I found that we can use the `export_named_datatype` function it provides. 

This will also fix the problem in my comment https://github.com/specta-rs/rspc/pull/323#issuecomment-2510572276

```rs
for export in self.type_map.iter().map(|(_, ty)| {
    export_named_datatype(
        &config,
        ty,
        &self.type_map,
    )
    .unwrap()
}) {
    writeln!(file, "\n{}", export)?;
}
```